core/modules/util/DbgPrintHelper.h

```diff
@@ -0,0 +1,224 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+/**
+ * @file
+ *
+ * @ingroup util
+ *
+ * @brief Helper class for printing an object with a `dbgPrint` method to an ostream using the insertion
+ *        operator inline. For example `myostream << DbgPrintHelper(myObject)`
+ *        This is useful for cases where the ostream operator<< has been used for object serialization (as is
+ *        the case with the query objects, where it writes the query IR back to an SQL query string using
+ *        std::ostream& operator<<)
+ *
+ *        Note for efficiency the object does not keep its own copy of the passed-in shared pointer (thus
+ *        eliminating the reference count change mechanics. DO NOT KEEP AN INSTANTIATED COPY OF THIS CLASS;
+ *        the shared_ptr may be released. Use the DbgPrintHelper in the generation of on ostream and then
+ *        throw it away.
+ *
+ *        A nullptr may be passed into the constructor, in that case 'nullptr' will be printed out.
+ *
+ * @author Nathan Pease, SLAC
+ */
+
+#ifndef LSST_QSERV_UTIL_DBG_PRINT_HELPER_H
+#define LSST_QSERV_UTIL_DBG_PRINT_HELPER_H
+
+#include <deque>
+#include <memory>
+#include <ostream>
+#include <vector>
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+
+template <typename T>
+class DbgPrintH {
+public:
+    DbgPrintH(const T& dumpee)
+    : _dumpee(dumpee) {}
+
+    const T& _dumpee;
+};
+
+template <typename T>
+std::ostream& operator<<(std::ostream& os, const DbgPrintH<T>& helper) {
+    helper._dumpee.dbgPrint(os);
+    return os;
+}
+
+
+template <typename T>
+class DbgPrintPtrH {
+public:
+    DbgPrintPtrH(const std::shared_ptr<T>& dumpee)
+    : _dumpee(dumpee) {}
+
+    const std::shared_ptr<T>& _dumpee;
+};
+
+template <typename T>
+std::ostream& operator<<(std::ostream& os, DbgPrintPtrH<T> const& dbgPrintHelper) {
+    if (dbgPrintHelper._dumpee != nullptr) {
+        dbgPrintHelper._dumpee->dbgPrint(os);
+    } else {
+        os << "nullptr";
+    }
+    return os;
+}
+
+
+template <typename T>
+class DbgPrintVectorH {
+public:
+    DbgPrintVectorH(const std::vector<T>& dumpees)
+    : _dumpees(dumpees) {}
+
+    const std::vector<T>& _dumpees;
+};
+
+template <typename T>
+std::ostream& operator<<(std::ostream& os, DbgPrintVectorH<T> const& helper)  {
+    os << "(";
+    for (const T & dumpee : helper._dumpees) {
+        os << dumpee;
+        if (dumpee != helper._dumpees.back()) {
```

Comment:
I think this repeats some logic that we already have in util/IterableFormatter, maybe we can merge them together?
This `if` compares values, it will probably mess the output if vector contains other items that are the same as the last item?